chat-android/src/main/java/com/ably/chat/MessagesReactions.kt

```diff
@@ -0,0 +1,532 @@
+    override val reaction: MessageReaction,
+) : MessageReactionRawEvent
+
+internal data class DefaultMessageReactionSummaryEvent(
+    override val summary: MessageReactionsSummary,
+    override val type: MessageReactionSummaryEventType = MessageReactionSummaryEventType.Summary,
+) : MessageReactionSummaryEvent
+
+internal data class DefaultMessageReaction(
+    override val messageSerial: String,
+    override val type: MessageReactionType,
+    override val name: String,
+    override val count: Int?,
+    override val clientId: String,
+) : MessageReaction
+
+internal class DefaultMessagesReactions(
+    private val chatApi: ChatApi,
+    private val roomId: String,
+    private val channel: Channel,
+    private val annotations: RealtimeAnnotations,
+    private val options: MessageOptions,
+    parentLogger: Logger,
+) : MessagesReactions {
+
+    private val logger = parentLogger.withContext("MessagesReactions")
+
+    private val reactionsScope = CoroutineScope(Dispatchers.Default.limitedParallelism(1) + SupervisorJob())
+
+    private val listeners: MutableList<MessagesReactions.Listener> = CopyOnWriteArrayList()
+    private val rawEventlisteners: MutableList<MessagesReactions.MessageRawReactionListener> = CopyOnWriteArrayList()
+
+    private val summaryEventBus = MutableSharedFlow<MessageReactionSummaryEvent>(
+        extraBufferCapacity = 1,
+        onBufferOverflow = BufferOverflow.DROP_OLDEST,
+    )
+
+    private val rawEventBus = MutableSharedFlow<MessageReactionRawEvent>(
+        extraBufferCapacity = UNLIMITED,
+        onBufferOverflow = BufferOverflow.DROP_OLDEST,
+    )
+
+    private val summarySubscription: Subscription
+    private var annotationSubscription: Subscription? = null
+
+    init {
+        reactionsScope.launch {
+            summaryEventBus.collect { summaryEvent ->
+                listeners.forEach {
+                    it.onReactionSummary(summaryEvent)
+                }
+            }
+        }
+
+        reactionsScope.launch {
+            rawEventBus.collect { rawEvent ->
+                rawEventlisteners.forEach {
+                    it.onRawReaction(rawEvent)
+                }
+            }
+        }
+
+        val messageSummaryListener = PubSubMessageListener {
+            internalMessageSummaryListener(it)
+        }
+
+        channel.subscribe(messageSummaryListener)
+        summarySubscription = Subscription { channel.unsubscribe(messageSummaryListener) }
+
+        if (options.rawMessageReactions) {
+            val annotationListener = RealtimeAnnotations.AnnotationListener {
+                internalAnnotationListener(it)
+            }
+            annotations.subscribe(annotationListener)
+            annotationSubscription = Subscription { annotations.unsubscribe(annotationListener) }
+        }
+    }
+
+    override suspend fun send(
+        messageSerial: String,
+        name: String,
+        type: MessageReactionType?,
+        count: Int,
```

Comment:
As per CHA-MR4b3, `count` must be a positive integer. So maybe throw clientError if `count <=0`

Reply:
added both checks